src/Dhall/TypeCheck.hs

```diff
@@ -444,6 +444,13 @@ typeWith _      OptionalFold      = do
                 (Pi "optional" (Const Type)
                     (Pi "just" (Pi "_" "a" "optional")
                         (Pi "nothing" "optional" "optional") ) ) ) )
+typeWith _      OptionalBuild     = do
+    return
+        (Pi "a" (Const Type)
+            (Pi "f" f (App Optional "a") ) )
```

Comment:
Minor suggestion: replace `"f"` with `"_"` for consistency with `List/build` so that it doesn't show up in the type